PHP/GenerateE2Authcode.php

```diff
@@ -0,0 +1,49 @@
+<?php
+function generateParamsIn(array $paymentParameters): string
+{
+    unset($paymentParameters["MERCHANT_HASH"]);
+    return implode(",", array_keys($paymentParameters));
+}
+
+function generateAuthcode($paymentParameters): string
+{
+    $authcodeString = implode("|", array_values($paymentParameters));
+    return strtoupper(hash("sha256", $authcodeString));
+}
+
+function formatHtml($paymentParameters): void
+{
+    array_walk($paymentParameters, function ($value, $key) use (&$inputs) {
```

Comment:
In this case since I have string indexed array I cant access the `key` inside `array_map()`. And since I need both `key` and `value` to generate the HTML input field I landed upon this. Due to the concatenation of line break the output will contain one extra line break in the end, but it does not affect the functionality so didn't bother to trim it.

It could be done by passing reference to an array an imploding it:

```
    $inputs = [];
    array_walk($paymentParameters, function ($value, $key) use (&$inputs) {
        $inputs[] = "<input name=\"$key\" type=\"hidden\" value=\"$value\">";
    });
    $inputs = implode("\n", $inputs);
```

But somehow I found the string concatenation pleasing.